src/devices/Is31fl3730/Is31fl3730.cs

```diff
@@ -0,0 +1,383 @@
+    // Datasheet: https://cdn-shop.adafruit.com/product-files/3017/31FL3730.pdf
+    // Product: https://shop.pimoroni.com/products/microdot-phat
+    // Product: https://shop.pimoroni.com/products/led-dot-matrix-breakout
+    // Related repo: https://github.com/pimoroni/microdot-phat
+    // Based on: https://github.com/pimoroni/microdot-phat/blob/master/library/microdotphat/matrix.py
+    public class Is31fl3730
+    {
+        private readonly byte[] _matrix_addresses = new byte[] { Is31fl3730FunctionRegister.Matrix1, Is31fl3730FunctionRegister.Matrix2 };
+        private readonly List<byte[]> _buffers = new List<byte[]>
+        {
+            new byte[8],
+            new byte[8]
+        };
+        private readonly bool[] _enabled = new bool[2];
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// Initialize IS31FL3730 device.
+        /// </summary>
+        /// <param name="i2cDevice">The <see cref="System.Device.I2c.I2cDevice"/> to create with.</param>
+        public Is31fl3730(I2cDevice i2cDevice)
+        {
+            _i2cDevice = i2cDevice ?? throw new ArgumentNullException(nameof(i2cDevice));
+        }
+
+        /*
+        The Pimoroni devices support three I2C devices:
+
+            - 0x61 (the default)
+            - 0x62
+            - 0x63
+
+        For the breakout, this is straightforward. There is a default and you can change the address (in the normal way).
+
+        For the Micro Dot pHAT, there are three pairs installed (making six matrices). The addresses are ordered this way:
+
+           x63       x62        x61
+        ________  _________  ________
+        m2 | m1 || m2 | m1 || m2 | m1
+
+        This ordering makes sense if you are scrolling content right to left.
+
+        More detailed information about the structure of each pair follows (further down).
+        */
+
+        /// <summary>
+        /// Default I2C address for device.
+        /// </summary>
+        public const int DefaultI2cAddress = 0x61;
+
+        /// <summary>
+        /// Enables or disables auto-buffering.
+        /// </summary>
+        public bool BufferingEnabled { get; set; } = true;
+
+        /// <summary>
+        /// Indexer for matrices.
+        /// </summary>
+        public DotMatrix5x7 this[int index] => GetMatrix(index);
+
+        /// <summary>
+        /// Update bright of LED matrix for each of 128 items.
+        /// </summary>
+        // From datasheet:
+        // Set high bit for the 128th item
+        // Set high bit low for the remaining 127 items
+        // And then set lower bits as appropriate, to represent 0-127
+        public void UpdateBrightness(byte value) => Write(Is31fl3730FunctionRegister.Pwm, value);
+
+        /// <summary>
+        /// Update current setting for each row output of LED matrix (default value is 40 mA).
+        /// </summary>
+        public void UpdateCurrent(Current value) => Write(Is31fl3730FunctionRegister.LightingEffect, (byte)value);
```

Comment:
While we're not doing that everywhere correctly, it would be good to check for valid values here as well. The user could pass any value that is not part of the enum.